canonical-web-and-design/ubuntu.com
# Advantage shop guest checkout

## Done

- Adds ability for non logged in users to make purchases
- Also removes the need for logged in users who aren't already an admin on an enterprise account to have to be manually setup as one.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- Make sure you're logged out by visiting https://login.ubuntu.com/+logout
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage/subscribe
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html
- See that the page header informs you that you may want to sign in if you have an existing account, but _do not do that_.
- Use the product selection form to select a product, add it to the cart, and click "Buy Now"
- Fill out the form with a plus variant of an email address you have access to e.g. scott.mason.nash<b>+guest</b>@canonical.com
  - You can also attempt to use your Ubuntu SSO email address. When clicking "Continue" in that case, you should see an error notification prompting you to log in.
- Use 4242 4242 4242 4242 as the card number, 04/24 as expiry, 242 as CVC, and 42424 as ZIP
- Proceed through the form, to the second stage, accept terms and conditions, then click Pay.
- You should be directed to /advantage/subscribe/thanks, and prompted to create an SSO account with the email address you filled out.
- Follow the link to Ubuntu SSO and create your account (you may need to verify your email).
- When the account has been created, visit /advantage and see that the product you purchased is there.
